server/src/test/java/org/elasticsearch/monitor/jvm/HotThreadsTests.java

```diff
@@ -0,0 +1,291 @@
+
+    private List<StackTraceElement> makeThreadStackHelper(List<String[]> names) {
+        return names.stream().map(e -> {
+            // Cannot mock StackTraceElement because it's final
+            return new StackTraceElement(e[0], e[1], "Some_File", 1);
+        }).collect(Collectors.toList());
+    }
+
+    public void testIdleThreadsDetection() {
+        for (var threadName : new String[] { "Signal Dispatcher", "Finalizer", "Reference Handler" }) {
+            ThreadInfo mockedThreadInfo = mock(ThreadInfo.class);
+            when(mockedThreadInfo.getThreadName()).thenReturn(threadName);
+            assertTrue(HotThreads.isIdleThread(mockedThreadInfo));
+        }
+
+        for (var threadName : new String[] { "Notification Thread", "Common-Cleaner" }) {
+            ThreadInfo mockedThreadInfo = mock(ThreadInfo.class);
+            when(mockedThreadInfo.getThreadName()).thenReturn(threadName);
+            when(mockedThreadInfo.getStackTrace()).thenReturn(new StackTraceElement[0]);
+            assertFalse(HotThreads.isIdleThread(mockedThreadInfo));
+        }
+
+        var testJvmStack = makeThreadStackHelper(
+            List.of(
+                new String[]{"org.elasticsearch.monitor.test", "methodOne"},
+                new String[]{"org.elasticsearch.monitor.testOther", "methodTwo"}
+            ));
+
+        ThreadInfo notIdleThread = mock(ThreadInfo.class);
+        when(notIdleThread.getThreadName()).thenReturn("Not Idle Thread");
+        when(notIdleThread.getStackTrace()).thenReturn(testJvmStack.toArray(new StackTraceElement[0]));
+
+        assertFalse(HotThreads.isIdleThread(notIdleThread));
+
+        var idleThreadStackElements = makeThreadStackHelper(
+            List.of(
+                new String[]{"java.util.concurrent.ThreadPoolExecutor", "getTask"},
+                new String[]{"sun.nio.ch.SelectorImpl", "select"},
+                new String[]{"org.elasticsearch.threadpool.ThreadPool$CachedTimeThread", "run"},
+                new String[]{"org.elasticsearch.indices.ttl.IndicesTTLService$Notifier", "await"},
+                new String[]{"java.util.concurrent.LinkedTransferQueue", "poll"}
+            ));
+
+        for (var extraFrame : idleThreadStackElements) {
+            ThreadInfo idleThread = mock(ThreadInfo.class);
+            when(idleThread.getThreadName()).thenReturn("Idle Thread");
+            when(idleThread.getStackTrace()).thenReturn(new StackTraceElement[] {extraFrame});
+            assertTrue(HotThreads.isIdleThread(idleThread));
+
+            var topOfStack = new ArrayList<>(testJvmStack);
+            topOfStack.add(0, extraFrame);
+            assertIdleThreadHelper(idleThread, topOfStack);
+
+            var bottomOfStack = new ArrayList<>(testJvmStack);
+            bottomOfStack.add(extraFrame);
+            assertIdleThreadHelper(idleThread, bottomOfStack);
+
+            var middleOfStack = new ArrayList<>(testJvmStack);
+            middleOfStack.add(1, extraFrame);
```

Comment:
Maybe use a random position: `middleOfStack.add(between(0, testJvmStack.size()), extraFrame);`?

Reply:
Good idea! I'll set it up such that it's random between start + 1, end-1 so that I don't accidentally duplicate what I'm already testing with top and bottom.